### ISSUE TITLE
Minimum fan speed

### ISSUE THREAD

**Issue by phu9Suop** [TheAomx/HP-625-Linux-Fancontrol#3]
Hi 
Thanks for the fan control software.
I see that there are a number of high temperatures inside various components of the hp625, before the fan starts operating . 
Is it possible to set a small minimum speed with this SW, to keep temperature low in normal operation?

Best regards
phu9Suop

**Comment by TheAomx** [comment]
Hi,
you can adjust the values of lower_bound and upper_bound in main.cpp to activate the fans on lower temperatures.

Currently it is configured like this:

> const std::array<ThermalArea, 4> thermal_areas = {{
>       {.lower_bound = 67, .upper_bound = 110, .devices = all_fans_on},
>       {.lower_bound = 60, .upper_bound = 67,  .devices = middle_intensity_fans},
>      {.lower_bound = 50, .upper_bound = 60,  .devices = small_intensity_fans},
>       {.lower_bound = 30, .upper_bound = 50,  .devices = all_fans_off},
> }};


Which means that the fans are activated at the threshold of 50 degree celcius. But you are free to adjust that threshold like you want to. Then the fans will be activated at an lower temperature. 

If they dont activate on 50 degree, then maybe you have to adjust the following line in main.cpp:

`    std::string command = R"foo(acpi -t | grep 'Thermal 0'| grep -oEe [0-9]+\.[0-9])foo";`

Maybe your cpu temprature is not listed at "Thermal 0". Try some other device in that case. You can always try to run in a terminal the following command:

> acpi -t | grep 'Thermal 0'| grep -oEe [0-9]+\.[0-9]

It should print the current temperature of your cpu! If not try "Thermal 1" and see if that works. What version of the linux kernel are you using?

**Comment by phu9Suop** [comment]
Hi 
Changing in the foo string to "grep 'Thermal 1'" and recompiling lowers the temperature. Speed reduction now begins at 70 °C and is not ON/OFF anymore. The whole NB runs cooler now < 69°C and with thermally stable operating points. Left Hand rest is not hot any more. That is desirable. 
my kernel version : uname -a
Linux hp625 4.18.0-0.bpo.1-amd64 #1 SMP Debian 4.18.6-1~bpo9+1 (2018-09-13) x86_64 GNU/Linux
Hardware Data 
Model HP 625 
SKN NUmmer xxxxxxxx
Notebook Seriennummer xxxxxxxxx
System ID 1475
ROM Revision 68DVA Ver F.06
ROM Datum 12/07/2010
Grafik BIOS Revision ATI03/18/10
Tastaturkontroller rev. 72.0E
Prozessortyp AMD Athlon II P360 Dual Core
2300 MHz
Speicherkap 8192 MByte 
Seriennummer Akku 1 xxxxx  2/10/2011
WLAN FCC ID QDS-BRCM1051
Bluetooth FCC ID QDS-BRCM1051

It solved the issue
Thank you for the quick answer. Best Regards
Erich

**Comment by TheAomx** [comment]
Great that this worked out for you!

**Comment by phu9Suop** [comment]
Hi TheAomx
Now I changed the kernel from Backports back to the stable version to have a stable only system.
It broke HP-625-Linux-Fancontrol.
I had to purge and  redo the whole installation procedure as in README.md. 
uname -a
Linux hp625 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux.
It was not a big deal , and everything worked as described.
However there is a conflict with the package in debian stable fancontrol (1:3.4.0-4) , which must not be installed on a hp625 IMHOP. 
This is for the record only, because I installed it during trouble shooting and I had to remove and purge  it again. 
What are the chances to automatically do a reinstallation of HP-625-Linux-Fancontrol or actualization on kernel updates ? I see this coming, as I do weekly security updates. 
Thx again
Erich

**Comment by TheAomx** [comment]
Hi,
did i understand you correclty that by upgrading the kernel you had to reinstall the fancontrol service? 

This would be weird, because normally Systemd-Services will not be removed by a kernel update (at least at my arch linux based system). So therefore if the installed executable under the folder /opt/ is still after the upgrade installed (which it should be) and the systemd-service is enabled after the kernel upgrade (which it also should be) then it can only be that the newer kernels requires some modification in the fan control program.

If you take a look at https://github.com/TheAomx/HP-625-Linux-Fancontrol/blob/master/main.cpp at line 33 there is a configuration for linux kernels lower than 4.13 and at line 42 is a configuration for kernels higher than 4.13. You seem to be on kernel release 4.9.x => maybe you should activate the configuration for kernels lower than 4.13.

Because i use ArchLinux at my HP625 there will always be default configurations for the latest kernel revisions by default. I will keep the old settings in the programm, but every user has to select the right settings for the installed kernel manually at the moment.

greetings,
Aomx

**Comment by phu9Suop** [comment]
Hi
the initial issue was that I downgraded from 4.13. to 4.9, to not have a mixed installation. All is stable at present. 
with kernel 4.13 on my hp625 the cpu temperature probe was at thermal 1
with kernel 4.09 on my hp625 the cpu temperature probe is     at thermal 0
after this change I had strange temperatures again- as usual up to 80 °C and then the bios emergency protection acted with full speed.
In addition and for historic reasons I shifted the executable to /usr\*bin, because /usr used to be a separate partition. This seems to have created a conflict with the dpkg fancontrol . /opt used to be on a very small root partition.  I once filled a root -filesystem / to the limit with a large application in /opt/ , which made the station rather difficult to repair. So i learnt to avoid having sth in /opt/. 
Now this hp625 dos not have this anymore. 
So this time with the second installation i did not modify anything from your setup. 
That and your Information should let hope that an upgrade does not break something in the future. 
Are there thoughts to make a debian/ubuntu/* package from this ? 
Thank you for your prompt reply.
Erich
